### PR TITLE
feat: add event emission hints with phase-aware middleware injection

### PR DESCRIPTION
**Pull request by rsalus**
## Summary

Adds a `check_event_emissions` orchestrate action that computes expected-but-missing model-emitted events per workflow phase, and injects `_eventHints` into MCP tool responses via the telemetry middleware. This gives the model actionable nudges when it should be emitting events.

- `PHASE_EXPECTED_EVENTS` registry mapping 6 workflow phases to expected model-emitted event types
- `handleCheckEventEmissions` handler with gate event emission and structured hint output
- `injectEventHints` in middleware following the existing `_autoCorrection` injection pattern
- Registered as action #17 in the orchestrate composite

## Changes

- **`check-event-emissions.ts`** (NEW) — Phase registry, human-readable event descriptions, handler with materializer + stream query
- **`check-event-emissions.test.ts`** (NEW) — 12 tests: registry validation, handler edge cases, gate emission, composite integration
- **`composite.ts`** — Registered `check_event_emissions` handler
- **`registry.ts`** — Added `check_event_emissions` action schema (action #17)
- **`registry.test.ts`** — Updated action count from 16 to 17
- **`middleware.ts`** — Added `injectEventHints` function, wired into `withTelemetry` HOF
- **`middleware.test.ts`** — 3 new tests for hint injection (with hints, empty, non-JSON)
- **`store.ts`** — Added `get dir()` public getter for stateDir access

## Test Plan

- [x] `PHASE_EXPECTED_EVENTS` entries are all model-emitted (compile-time assertion + test)
- [x] Handler returns hints for missing events, empty for complete/unknown phases
- [x] Gate emission failure doesn't break handler (fire-and-forget)
- [x] `_eventHints` injected into JSON responses, non-JSON responses unchanged
- [x] All 3335 MCP server tests pass
- [x] Typecheck clean

---

Stack 2/2 of model-emitted event reliability refactor (stacked on #955). Related: #952

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: servers/exarchos-mcp/src/telemetry/middleware.ts [hunk 1]
```diff
@@ -70,6 +70,32 @@ function injectAutoCorrection(result: McpToolResult, applied: Correction[]): Mcp
...
+/** Injects `_eventHints` into the JSON payload of the first content entry. Fails silently if text is not valid JSON. */
+function injectEventHints(result: McpToolResult, hints: EventHint[]): McpToolResult {
+  if (hints.length === 0) return result;
+
+  const entry = result.content[0];
+  if (!entry?.text) return result;
+
+  try {
+    const parsed = JSON.parse(entry.text) as Record<string, unknown>;
+    parsed._eventHints = { missing: hints, phase: 'unknown', checked: hints.length };
```
[the remark this answers is not in the document]
Comment (rsalus) [reply servers/exarchos-mcp/src/telemetry/middleware.ts]: Fixed in 5ae30e91 — injectEventHints now accepts an EventHintsPayload with real phase and checked from the handler result.

File: servers/exarchos-mcp/src/orchestrate/check-event-emissions.ts [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+  if (!args.featureId) {
+    return {
+      success: false,
+      error: { code: 'INVALID_INPUT', message: 'featureId is required' },
+    };
+  }
+
+  const store = getOrCreateEventStore(stateDir);
+  const materializer = getOrCreateMaterializer(stateDir);
+  const streamId = args.workflowId ?? args.featureId;
```
[the remark this answers is not in the document]
Comment (rsalus) [reply servers/exarchos-mcp/src/orchestrate/check-event-emissions.ts:104]: Fixed in 5ae30e91 — added SAFE_STREAM_ID regex validation for both featureId and workflowId with corresponding tests.

File: servers/exarchos-mcp/src/telemetry/middleware.ts [hunk 3]
```diff
@@ -70,6 +70,32 @@ function injectAutoCorrection(result: McpToolResult, applied: Correction[]): Mcp
...
+    return {
+      ...result,
+      content: [{ ...entry, text: JSON.stringify(parsed) }, ...result.content.slice(1)],
+    };
```
[the remark this answers is not in the document]
Comment (rsalus) [reply servers/exarchos-mcp/src/telemetry/middleware.ts]: Fixed in 5ae30e91 — injectEventHints now takes an EventHintsPayload interface. The caller passes phase and checked from the actual handler result.

File: servers/exarchos-mcp/src/telemetry/middleware.ts [hunk 4]
```diff
@@ -177,6 +203,24 @@ export function withTelemetry(
...
+            eventStore.dir,
+          );
+          if (hintResult.success && hintResult.data) {
+            const data = hintResult.data as { hints?: EventHint[]; phase?: string; checked?: number };
+            if (data.hints && data.hints.length > 0) {
+              finalResult = injectEventHints(finalResult, data.hints);
+            }
+          }
+        } catch { /* fire-and-forget — hint generation failure never blocks */ }
+      }
```
[the remark this answers is not in the document]
Comment (rsalus) [reply servers/exarchos-mcp/src/telemetry/middleware.ts]: Fixed in 5ae30e91 — hint generation now uses Promise.race with a 150ms timeout. If the hint check exceeds the budget, the response returns without hints.
